sentry-core/src/performance.rs

```diff
@@ -660,15 +672,22 @@ impl<'a> TransactionData<'a> {
 impl Transaction {
     #[cfg(feature = "client")]
     fn new(client: Option<Arc<Client>>, ctx: TransactionContext) -> Self {
-        let (sampled, transaction) = match client.as_ref() {
+        let ((sampled, sample_rate), transaction) = match client.as_ref() {
             Some(client) => (
-                client.is_transaction_sampled(&ctx),
+                client.determine_sampling_decision(&ctx),
                 Some(protocol::Transaction {
                     name: Some(ctx.name),
                     ..Default::default()
                 }),
             ),
-            None => (ctx.sampled.unwrap_or(false), None),
+            None => (
+                (
+                    ctx.sampled.unwrap_or(false),
+                    ctx.sampled
+                        .map_or(0.0, |sampled| if sampled { 1.0 } else { 0.0 }),
```

Comment:
[`f32::from`](https://doc.rust-lang.org/std/primitive.f32.html#method.from-6) does this conversion already

```suggestion
                        .map_or(0.0, f32::from),
```